### PR TITLE
Add differentiable system-identification worked example

### PR DESCRIPTION
**Pull request by jslee02**
## Summary

- Adds a standalone, torch-free Python worked example,
  `python/examples/diff_system_identification.py`, that recovers an unknown
  rigid-body **mass** by plain gradient descent on DART's opt-in differentiable
  simulation API. Researchers get a runnable, end-to-end system-identification
  reference for the differentiable `World` (PLAN-110) that needs no PyTorch.
- The script runs in the default build too: when differentiable support is not
  compiled (`DART_BUILD_DIFF=OFF`, the default pixi build) it prints a clear
  message and exits 0 instead of raising.

## Motivation / Problem

The differentiable-simulation surface (analytic step derivatives, parameter
Jacobians, VJP, dartpy/torch bridges) landed in PR #2761, and GUI
trajectory-optimization scenes already ship. What was still missing for the
DART 8 promotion contract was (a) a standalone *example program* (not a GUI
`demos/` scene) and (b) any system-identification example in any form. This is
the one differentiable workflow with zero prior example. The example doubles as
living documentation of the `add_differentiable_parameter` /
`get_step_derivatives` parameter-gradient path.

## Changes / Key Changes

- New `python/examples/diff_system_identification.py`:
  - Builds a tiny differentiable `World` with a single rigid body driven by a
    known constant force (zero gravity, contact-free, so the trajectory depends
    on the mass only through `F/m` and the smooth differentiable step is exact).
  - Generates a ground-truth trajectory at the planted true mass, then recovers
    the mass from a deliberately wrong guess via an adjoint reverse pass over the
    chained per-step `state_jacobian` / `parameter_jacobian` (mirrors the
    reverse pass in `diff_cartpole_trajopt._optimize_cartpole`).
  - Prints iteration / loss / mass / gradient and a PASS line; the analytic
    gradient is consistent with the closed form `dv'/dm = -dt F / m^2`.
  - Graceful degradation: detection via `sx.diff.rollout` presence plus a
    guarded `add_differentiable_parameter` / `get_step_derivatives` probe; on
    unavailability it prints a message and `sys.exit(0)` (no GUI fallback).
- `CHANGELOG.md`: entry under *GUI, Examples, and Tutorials*.
- `docs/dev_tasks/differentiable_simulation/{README,RESUME}.md`: mark the
  system-identification worked example as shipped; the standalone
  trajectory-optimization example remains the open promotion follow-up.

## Before / After

| Dimension | Before | After |
| --- | --- | --- |
| Examples (commands) | GUI trajopt scenes via `py-demos`; no standalone diff script; no system-ID example | `python python/examples/diff_system_identification.py` runs end-to-end (recovers mass on a DIFF-ON build; prints a graceful message on the default DIFF-OFF build) |
| Migration / opt-in | n/a | No new pixi task, no CMake edit; run directly with `PYTHONPATH=build/default/cpp/Release/python:python` |

## Testing

- `pixi run lint` — green.
- `pixi run test-py` (DIFF-ON build via `DART_BUILD_DIFF_OVERRIDE=ON`) —
  **1437 passed, 11 skipped** (the only diff-related skip is the torch autograd
  test, `test_diff.py:481`, since torch is not installed in-env).
- DIFF-OFF (default build) run — exits 0 with the graceful "differentiable
  simulation is not enabled in this build" message:
  `PYTHONPATH=build/default/cpp/Release/python:python .pixi/envs/default/bin/python python/examples/diff_system_identification.py`
- DIFF-ON build (`DART_BUILD_DIFF_OVERRIDE=ON pixi run build-py-dev`) run —
  recovers the planted **2.0 kg** mass from a 5.0 kg guess in 23 iterations
  (relative error ~2.3e-7, final loss ~4.3e-13), prints converged
  iteration/loss/mass, `PASS`, exits 0.

## Breaking Changes

- [x] None
- Additive Python example plus dev_task doc and changelog edits; no C++, no
  bindings, no pixi task, no CMake changes.

## Related Issues / PRs (backports)

- Builds on #2761 (opt-in differentiable simulation, PLAN-110). No backport
  (DART 7 / `main` only). The standalone trajectory-optimization example is a
  tracked follow-up, intentionally out of scope here.

---

#### Checklist

- [x] Milestone set: DART 7.0 (`main`)
- [x] CHANGELOG.md updated per `docs/onboarding/changelog.md` (the PR-number
      link is added as an approved follow-up once this PR number exists)
- [x] N/A — no new library functionality; the example is itself a runnable
      verification and the differentiable API it exercises is covered by
      `python/tests/unit/simulation/test_diff.py`
- [x] N/A — no new public methods or classes (the example is self-documented)
- [x] N/A — no dartpy binding changes

### CODE REVIEW

**Comment by jslee02** [comment]
@codex review

**Comment by jslee02** [comment]
@codex review

**Comment by jslee02** [comment]
@codex review
